### PR TITLE
Add factory methods, tests and messages for Changes.

### PR DESCRIPTION
**Pull request by alexander-aleksandrov**

### CODE REVIEW

File: client/src/main/proto/spine/base/change.proto [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+}
+
+// Definition of a change in a `bool` field.
+message BooleanChange {
+
+    // The value of the field that's changing.
+    bool previous_value = 1;
+
+    // The new value of the field.
+    bool new_value = 2 [(required) = true];
```
Comment (alexander-yevsyukov) [inline client/src/main/proto/spine/base/change.proto]: How can it be missing?

File: client/src/main/java/org/spine3/base/Changes.java [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+
+    /**
+     * Creates {@link StringChange} object for the passed previous and new field values.
+     *
+     * <p>Passed values cannot be equal.
+     */
+    public static StringChange of(String previousValue, String newValue) {
+        checkNotNull(previousValue, PREVIOUS_VALUE);
+        checkNotNull(newValue, NEW_VALUE);
+        checkArgument(!newValue.isEmpty(), "newValue cannot be empty");
```
Comment (alexander-yevsyukov) [inline client/src/main/java/org/spine3/base/Changes.java]: Please add test for this line and all those that appear yellow (partially tested).

Comment (alexander-yevsyukov) [inline client/src/main/java/org/spine3/base/Changes.java]: Resolved. We have tests. It's a plug-in bug.

File: client/src/main/proto/spine/base/change.proto (hunk 1 above)
Comment (alexander-aleksandrov) [inline client/src/main/proto/spine/base/change.proto]: Fixed

**Pull request merged**